src/plugins/registry.js

```diff
@@ -74,8 +74,25 @@ module.exports = function (bookshelf) {
   // can't include it with the rest of the relational methods.
   const morphTo = Model.prototype.morphTo;
   Model.prototype.morphTo = function(relationName) {
-    return morphTo.apply(this, [relationName].concat(map(drop(arguments), (model) => {
-      return resolveModel(model);
+    let candidates, columnNames;
+    if (isArray(arguments[1])) {
+      columnNames = arguments[1];
+    } else {
+      columnNames = null;
+    }
+    if (isArray(arguments[1]) || isNil(arguments[1])) {
+      candidates = drop(arguments, 2);
+    } else {
+      candidates = drop(arguments, 1);
+    }
```

Comment:
This could be optimized to
```js
if (isArray(arguments[1]) || isNil(arguments[1])) {
  columnNames = arguments[1]; // may be `null`
  candidates = drop(arguments, 2);
} else {
  columnNames = null;
  candidates = drop(arguments, 1);
}
```